contractcourt/chain_watcher.go

```diff
@@ -691,10 +691,36 @@ func (c *chainWatcher) dispatchCooperativeClose(commitSpend *chainntnfs.SpendDet
 	log.Infof("Cooperative closure for ChannelPoint(%v): %v",
 		c.cfg.chanState.FundingOutpoint, spew.Sdump(broadcastTx))
 
+	// Reload the channel state in memory to obtain the latest channel
+	// status. This is required to determine the close initiator of the
+	// channel.
+	if err := c.cfg.chanState.RefreshChanInfo(); err != nil {
```

Comment:
Perhaps a better place for this would be within `closeObserver` right after we receive the spend notification so that it applies to all cases? I'm not too familiar with all the areas that can modify our channel state, but we should be careful of this change in general.

Reply:
Yeah, we seem to follow this pattern in a bunch of places in chainwatcher; if we need a fresh value we re-query db then mutate those values in memory. 

Had a look at this one specifically, and the only fields that would change on this call are `c.IsPending`, `c.chanStatus`, `c.TotalMsatSend` and `c.TotalMsatReceived` because we are only updating chan info. Of these fields, only `chanStatus` is used in `chainwatcher` so I think this should be ok (especially since we don't clear channel statuses, just write more bits). But I will move it up so that it's used across close types 👍